dascore/proc/basic.py

```diff
@@ -480,3 +481,89 @@ def dropna(patch: PatchType, dim, how: Literal["any", "all"] = "any") -> PatchTy
     cm = patch.coords.update(**{dim: coord[to_keep]})
     attrs = patch.attrs.update(coords={})
     return patch.new(data=new_data, coords=cm, attrs=attrs)
+
+
+@patch_function()
+def pad(
+    patch: PatchType,
+    mode: Literal["constant"] = "constant",
+    constant_values: Any = 0,
+    expand_coords=False,
+    samples=False,
+    **kwargs,
+) -> PatchType:
+    """
+    Pad the patch data along specified dimensions.
+
+    Parameters
+    ----------
+    mode : str, optional
+        The mode of padding, by default 'constant'.
+    constant_values : scalar or sequence, optional
+        If a scalar, this value is used as the pad value for
+        all dimensions. If a sequence, it must be of the same
+        length as the number of dimensions being padded,
+        and each element of the sequence
+
+        specifies the pad value for the corresponding dimension.
+    **kwargs:
+        Used to specify dimension and number of elements,
+        either an integer or a tuple (before, after).
+
+    Examples
+    --------
+    >>> import dascore as dc
+    >>> patch = dc.get_example_patch()
+    >>> # zero pad `time` dimension with 2 zeros before and 3 zeros after
+    >>> padded_patch_1 = patch.pad(time = (2, 3))
+    >>> # zero pad `distance` dimension with 4 unit values before and after
+    >>> padded_patch_2 = patch.pad(distance = 4, constant_values = 1)
+    """
+    pad_width = [(0, 0)] * len(patch.shape)
+    dimfo = get_multiple_dim_value_from_kwargs(patch, kwargs)
+
+    new_coords = {}
+    for _, info in dimfo.items():
+        axis, dim, value = info["axis"], info["dim"], info["value"]
+        # Ensure passed dimension exists
+        if dim not in patch.dims:
```

Comment:
You can remove this. `get_multiple_dim_value_from_kwargs` will ensure you only get dims back, then you will have full test coverage.

Reply:
Sorry, reviewing this from my phone and the wrong lines got selected. It should be 529 and 530 (the if and continue clause to skip non dim names)